lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/client/YarnSubmissionHelper.java

```diff
@@ -279,6 +282,13 @@ public void submit() throws IOException, YarnException {
     }
 
     this.yarnClient.submitApplication(applicationSubmissionContext);
+
+    if (this.isUnmanaged) {
+      // For Unmanaged AM mode, add a new app token to the
+      // current process so it can talk to the RM as an AM.
+      final Token<AMRMTokenIdentifier> token = this.yarnClient.getAMRMToken(this.applicationId);
+      this.tokenProvider.addTokens(UserCredentialSecurityTokenProvider.serializeToken(token));
```

Comment:
@jwang98052 *"Unmanaged AM"* is a term used everywhere in YARN, and refers to an Application Master that that runs not on one of YARN-managed containers. It is a set phrase in YARN terminology, and I would much prefer to stick to it. I agree that the word "Unmanaged" should be used only as part of the "Unmanaged AM" or "Unmanaged Driver" phrase, and never alone.